source/includes/connection-snippets/x509-snippets/java-connection.java

```diff
@@ -0,0 +1,27 @@
+
+// begin x509 connection
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.MongoCollection;
+import org.bson.Document;
+import org.bson.BsonDocument;
+
+public class X509Connection {
+
+    public static void main(String args[]) {
+        // Set system properties via commandline or programmatically
+        System.setProperty("javax.net.ssl.keyStore", "/etc/certs/mongodb/v3/client.keystore");
+        System.setProperty("javax.net.ssl.keyStorePassword", "<your_password>");
+
+        String uri = "mongodb+srv://<cluster-url>/test?authSource=$external&retryWrites=true&w=majority&authMechanism=MONGODB-X509";
```

Comment:
To make it clear what's really required for X509, consider breaking it up into two lines, with the required part on the first line and the rest on the second line:

    String uri = "mongodb+srv://<cluster-url>/test?authMechanism=MONGODB-X509" 
                 + "&authSource=$external&retryWrites=true&w=majority";

Reply:
Updated to `mongodb+srv://<cluster-url>/test?authMechanism=MONGODB-X509&authSource=$external&retryWrites=true&w=majority`